Vehicle Spawner: removed VehicleList.xml and added Open Wheel category

katsutosh opened the pull request:
- added vehicle category `Openwheel` to Spawner
- added missing vehicles to VehicleList.xml (b3095)

katsutosh committed:
add Openwheel vehicle category to spawner and update VehicleList.xml

katsutosh committed:
update VehicleList.xml

itsjustcurtis reviewed (commented):
Before I merge this, is the vehicle list fully necessary at this point? Menyoo quite happily pulls vehicles into the correct categories based on the vehicles.meta data, and I wonder if we can do it without using the list at all. 

Happy to approve the new OpenWheel category if you're able to split that to a new PR

katsutosh commented:
good catch, didn't noticed it.
Just checked it, the list is not necessary.
Can we remove the code for reading the `VehicleList.xml`?

New PR for the OpenWheel category will be added later.

itsjustcurtis commented:
if you make the edits and confirm it working in game, I can merge the PR if you add the commits to this

katsutosh committed:
remove unused code in ModelNames.cpp `PopulateVehicleModelsArray`

metoxys commented:
I recommend renaming to "Open Wheel" and also adding a category for the drift cars added in the Chop Shop DLC

katsutosh committed:
remove VehicleList.xml and remove read code for VehicleList.xml

katsutosh committed:
- split each vehicleclass in it's own category
- add missing categories to spawnmenu
- removed fake vehicle classes

katsutosh committed:
rename Openwheel to Open Wheel

katsutosh commented:
- removed VehicleList.xml stuff
- each vehicleclass has it's own category in spawnmenu

All changes are working ingame.

(the driftcars are in the same category as the non-drift version)

metoxys commented:
> (the driftcars are in the same category as the non-drift version)

Consider hardcoding the model names into a separate Drift category

katsutosh committed:
add IS_VEHICLE_GEN9_EXCLUSIVE_MODEL native

katsutosh committed:
add missing vehicle hashes to VehicleHash enum

katsutosh committed:
vehicle spawner:
- move drift vehicles (Chop Shop DLC) to extra drift category
- exclude Gen9 vehicles from list

itsjustcurtis commented:
I'm not a fan of hard-coding and would have preferred not moving the drift cars to their own category, however with R*s insistence on making them the same vehicle with a different handling, I'm not sure how we can easily differentiate the drift variants. 

It may be worth (in the future) adding a "drift conversion" function to those with that availability to menyoo customs. For now this is okay, merging.

metoxys commented:
(commenting here because this fork has no Issue or Discussion area)  
@itsjustcurtis Consider the fact that the drift cars literally have their own model and handling IDs. For instance, `drifttampa` is the Drift Tampa with drift tuning whereas `tampa2` is the Drift Tampa that was added years ago.

itsjustcurtis commented:
> Consider the fact that the drift cars literally have their own model and handling IDs.

This is precisely why I've accepted the PR tbh, I can't think of a better way for now without having the drift conversion, which in and of itself would need to be hard-coded

metoxys commented:
> > Consider the fact that the drift cars literally have their own model and handling IDs.
> 
> This is precisely why I've accepted the PR tbh, I can't think of a better way for now without having the drift conversion, which in and of itself would need to be hard-coded

Well, presumably, GTAO handles the "drift conversion" by just changing the model of the car the player is in while keeping the data about modifications. But that just moves the hardcoding to another place, namely the list of cars that can be drift-converted, and the mapping between the drift and non-drift cars.  
In my opinion, the least intrusive method is a separate "Drift" category in the spawner

vbelevcev commented:
The menu does not see most of the transport models, maybe you should not have deleted the list? And as for me it is more convenient when you can add to the file as a separate list, and you can categorize yourself) on the contrary, I would like to see additional lists, such as a list of effects and so on. yeah normal in-game names wouldn't hurt either) It would also be nice to sort by add-on number

katsutosh commented:
> The menu does not see most of the transport models

Which vehicles do you mean by transport models? 
I checked all categories and the list looked complete for me

itsjustcurtis commented:
> The menu does not see most of the transport models, maybe you should not have deleted the list? And as for me it is more convenient when you can add to the file as a separate list, and you can categorize yourself) on the contrary, I would like to see additional lists, such as a list of effects and so on. yeah normal in-game names wouldn't hurt either) It would also be nice to sort by add-on number

Can you raise an issue and be more specific when you do please. I agree that being able to categorise yourself may have been a useful feature, though I personally never used it, that may have been the original intent behind the list.

vbelevcev commented:
I use the cheat for online and its downloader asi there are conflicts, did not see the list of add-ons) all with the list is fine I stupil) can I throw you a few improvements?, I did for myself, but the cheat as if for all you finalize, and with open source and not like that upstart who complained I now on the last account gave restrictions

itsjustcurtis commented:
@SkyK0T as I say, either raise suggestions in the Issues tab or send as a PR to this repository and I can review. Anything that is designed to modify GTAO content or gameplay however will be rejected as that violates TOS

vbelevcev commented:
pulrequests don't work here( I'll throw some of them here, and rockstar doesn't care about asi files for a long time already, there are much more powerful cheats.

vbelevcev commented:
Full PedList 1.68 reorganized
[PedList.zip](https://github.com/itsjustcurtis/MenyooSP/files/13989185/PedList.zip)

itsjustcurtis commented:
> pulrequests don't work here( I'll throw some of them here, and rockstar doesn't care about asi files for a long time already, there are much more powerful cheats.

You are commenting on a pull request. I assure you they work. You need to fork from this repo to then PR back into it. Being honest though, such a flagrant disregard for the TOS is not something I really want involved in this project, regardless if R* will actually do anything about it. My integrity as a programmer does not want to violate that.
